hamilton/driver.py

```diff
@@ -69,7 +69,8 @@ class Variable:
 
     name: str
     type: typing.Type
-    tags: Dict[str, str] = field(default_factory=frozenset)
+    tags: Dict[str, str] = field(default_factory=dict)
+    is_external_input: bool = field(default=False)
```

Comment:
we use `UD` -- so should it be `is_user_defined` instead?

Reply:
If anything we should change that -- `user_defined` to me makes less sense than `external`. Also `UD` is confusing without a legend...

Reply:
but maybe `is_external_input` is fine 🤔

Reply:
Yep sure. I'm just thinking about language. @gravesee do you have any preference here?